codechecker_common/review_status_handler.py

```diff
@@ -84,61 +86,85 @@ def __parse_codechecker_review_comment(
+            for field in rule['filters']:
+                if field not in ReviewStatusHandler.ALLOWED_FILTERS:
                     raise ValueError(
-                        f"Format error in {self.__review_status_yaml}: field "
-                        f"'{field}' is not allowed. Available fields are: "
-                        f"{', '.join(ReviewStatusHandler.ALLOWED_FIELDS)}")
+                        f"Format error in {self.__review_status_yaml}: filter "
+                        f"'{field}' is not allowed. Available filters are: "
+                        f"{', '.join(ReviewStatusHandler.ALLOWED_FILTERS)}")
 
-            if 'review_status' not in item and 'ignore' not in item:
+            for field in rule['actions']:
+                if field not in ReviewStatusHandler.ALLOWED_ACTIONS:
+                    raise ValueError(
+                        f"Format error in {self.__review_status_yaml}: action "
+                        f"'{field}' is not allowed. Available actions are: "
+                        f"{', '.join(ReviewStatusHandler.ALLOWED_ACTIONS)}")
+
+            if 'review_status' not in rule['actions'] and \
+                    'ignore' not in rule['actions']:
                 raise ValueError(
                     f"Format error in {self.__review_status_yaml}: "
-                    f"'review_status' or 'ignore' is required in {item}.")
+                    f"'review_status' or 'ignore' is required in {rule}.")
 
-            if 'review_status' in item:
+            if 'review_status' in rule['actions']:
                 review_status_options = \
                     ', '.join(ReviewStatusHandler.REVIEW_STATUS_OPTIONS)
 
-                if item['review_status'] \
+                if rule['actions']['review_status'] \
                         not in ReviewStatusHandler.REVIEW_STATUS_OPTIONS:
                     raise ValueError(
                         f"Invalid review status field: "
-                        f"{item['review_status']} at {item} in "
+                        f"{rule['actions']['review_status']} at {rule} in "
                         f"{self.__review_status_yaml}. Available options are: "
                         f"{review_status_options}.")
 
-                if item['review_status'] == 'suppress':
-                    item['review_status'] = 'false_positive'
+                if rule['actions']['review_status'] == 'suppress':
+                    rule['actions']['review_status'] = 'false_positive'
+
+    def __validate_review_status_yaml_data(self):
+        """
+        This function validates the data read from review_status.yaml file and
+        raises ValueError with the description of the error if the format is
+        invalid.
+        """
+        if not isinstance(self.__data, dict) or '$version' not in self.__data:
+            raise ValueError(
+                f"{self.__review_status_yaml} should be a dictionary with "
+                "the key '$version'.")
```

Comment:
Can we do something better?
How about we
1. Emit a separate warning for the file not being a valid yaml file (why call it a "dict" in the first place?), and when the "version" key is not found
2. Give a hint: "hint: Currently the latest version of this format is '1.0'. Consider adding 'version: 1.0' to the top of `<file>.yaml`."